metal/memory.h

```diff
@@ -0,0 +1,81 @@
+/* Copyright 2019 SiFive, Inc */
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#ifndef METAL__MEMORY_H
+#define METAL__MEMORY_H
+
+#include <stdint.h>
+#include <stddef.h>
+
+/*!
+ * @file memory.h
+ *
+ * @brief API for enumerating memories
+ */
+
+struct _metal_memory_attributes {
+	int R : 1;
+	int W : 1;
+	int X : 1;
+	int C : 1;
+	int A : 1;
+};
+
+/*!
+ * @brief A handle for a memory
```

Comment:
Is it worthwhile to note that it is a memory 'block' in the description.
Memory is rather vague.